feat(131): implement GraalVM native image support

nancymuyeh opened the pull request.

nancymuyeh committed:
feat(131): implement GraalVM native image support

stephane-segning reviewed (commented):
Nice work. Please address tests and the CI too

nancymuyeh committed:
feat(131): move the native plugin into a native profile to ensure it runs only when explicitely activated with -Pnative

nancymuyeh committed:
Update Workflow to run on PR for testing

nancymuyeh committed:
Change graav base image version to the one suggested here https://www.graalvm.org/latest/getting-started/container-images/

nancymuyeh committed:
Add and activate maven wrapper to allow use of ./mvnw command

nancymuyeh committed:
feat(131): implement GraalVM native image support

nancymuyeh committed:
update CI/CD workflow file to run on pull request to the develop branch

nancymuyeh committed:
Added NativeHints for the AOT compiler

stephane-segning committed:
fix: graalvm build

nancymuyeh committed:
feat: native support and docker-compose updates

- update dockerfile with requred dependencies

- Added a getting started and spring native documentation

- Added docker compose with postgress and pgAdmin for local testing

- Update docker release workflow to tun only on merge

nancymuyeh committed:
Merge branch 'develop' into 131-implement-graalvm-native-image-support

nancymuyeh committed:
update develop.yaml

stephane-segning commented on Dockerfile:
```diff
@@ -1,25 +1,53 @@
+    cp /lib/x86_64-linux-gnu/libc.so.6 /deps/lib/x86_64-linux-gnu/ && \
+    cp /lib64/ld-linux-x86-64.so.2 /deps/lib64/
+
+RUN find /lib/x86_64-linux-gnu -name "*.so*" -exec cp {} /deps/lib/x86_64-linux-gnu/ \; || true
```
I can see x86_64 here. What about people on aarch64? Please create a follow up ticket for this.

stephane-segning commented on docker-compose.yaml:
```diff
@@ -0,0 +1,38 @@
+version: '3.8'
```
We're on docker compose v4. Here we don't need this key `version:`

stephane-segning reviewed (changes requested):
Good work. Please address these small changes.

nancymuyeh committed:
Merge branch 'develop' into 131-implement-graalvm-native-image-support

nancymuyeh committed:
Update docker related files to add support for multi-architechtire builds

nancymuyeh committed:
Merge branch '131-implement-graalvm-native-image-support' of github.com:ADORSYS-GIS/webank-pending-registration-service into 131-implement-graalvm-native-image-support

nancymuyeh committed:
Update dockerfile

nancymuyeh committed:
Update dockerfile

nancymuyeh committed:
Update dockerfile

nancymuyeh committed:
Merge branch 'develop' into 131-implement-graalvm-native-image-support

nancymuyeh committed:
Update dockerfile

nancymuyeh committed:
Merge branch '131-implement-graalvm-native-image-support' of github.com:ADORSYS-GIS/webank-pending-registration-service into 131-implement-graalvm-native-image-support

Arielpetit reviewed (dismissed):
Great

nancymuyeh committed:
Update docker-release file

stephane-segning commented on Dockerfile:
```diff
@@ -1,25 +1,91 @@
-CMD ["java", "-jar", "/webank-prs/prs-rest-server-0.0.1-SNAPSHOT.jar"]
\ No newline at end of file
+# Copy libraries (the target platform will determine the correct paths)
+ARG TARGETPLATFORM
```
You don't need `TARGETPLATFORM` here

stephane-segning commented on secrets.env:
- Rename this file to `secrets.env.template`
- Remove Jude's email as email & password
- Use MailDev as smtp server
- Show the user how to create private_key & publik_key

stephane-segning commented on .github/workflows/docker-release.yaml:
Check how tags are managed here and cheat it: https://github.com/vymalo/image-resizer/blob/main/.github/workflows/build.yml#L68

stephane-segning reviewed (changes requested):
Wonderful work! It's awesome! Let's refine it a little ;)

nancymuyeh committed:
remove secrets file

nancymuyeh committed:
update workflow with docker tags

nancymuyeh committed:
update workflow with docker tags

nancymuyeh committed:
Merge branch 'develop' into 131-implement-graalvm-native-image-support

Arielpetit reviewed (approved):
LGTM
